sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -775,4 +783,120 @@ export class ApplicationControllerService {
+  ): Promise<ApplicationData> {
+    const application =
+      await this.applicationService.getApplicationForSubmissionValidation(
+        applicationId,
+        studentId,
+      );
+    if (!application) {
+      throw new NotFoundException("Application not found.");
+    }
+    if (!application.programYear.active) {
+      throw new UnprocessableEntityException("Program year is not active.");
+    }
+    // Validate the values in the submitted application before submitting.
+    await this.validateSubmitApplicationData(payload);
+    const submissionResult =
+      await this.formService.dryRunSubmission<ApplicationData>(
+        application.programYear.formName,
+        payload.data,
+      );
+    if (!submissionResult.valid) {
+      throw new BadRequestException(
+        "Not able to create an application due to an invalid request.",
+      );
+    }
+    await this.offeringIntensityRestrictionCheck(
+      studentId,
+      submissionResult.data.data.howWillYouBeAttendingTheProgram,
+    );
+    return submissionResult.data.data;
+  }
+
+  /**
+   * Validates and updates the payload values with the offering values (where the server is the source of truth) before submitting.
+   * This is required to the values in the payload to be the same as the values in the offering and to prevent the user from modifying them.
+   * @param payload payload to create the application.
+   * @throws BadRequestException if the offering intensity type is invalid.
+   * @throws UnprocessableEntityException if the offering is not valid.
+   * @throws UnprocessableEntityException if the program is not active.
+   * @throws UnprocessableEntityException if the program is expired.
+   */
+  private async validateSubmitApplicationData(
+    payload: SaveApplicationAPIInDTO,
+  ): Promise<void> {
+    const isFulltimeAllowed = this.configService.isFulltimeAllowed;
+    // The check to validate the values for howWillYouBeAttendingTheProgram can be removed once the toggle for IS_FULL_TIME_ALLOWED is no longer needed
+    // and the types are hard-coded again in the form.io definition using the onlyAvailableItems as true.
+    if (
+      ![OfferingIntensity.fullTime, OfferingIntensity.partTime].includes(
+        payload.data.howWillYouBeAttendingTheProgram,
+      )
+    ) {
+      throw new BadRequestException("Offering intensity type is invalid.");
+    }
+    if (
+      !isFulltimeAllowed &&
+      payload.data.howWillYouBeAttendingTheProgram ===
+        OfferingIntensity.fullTime
+    ) {
+      throw new UnprocessableEntityException("Invalid offering intensity.");
+    }
+
+    // studyStartDate from payload is set as studyStartDate
+    let studyStartDate = payload.data.studystartDate;
+    let studyEndDate = payload.data.studyendDate;
+    if (payload.data.selectedOffering) {
+      const offering = await this.offeringService.getOfferingById(
+        payload.data.selectedOffering,
+        { programId: payload.data.selectedProgram },
+      );
```

Comment:
Not part of the PR or may be part of @sh16011993 work, we should also check the match for Application `Offering Intensity` vs `Offering Intensity` from actual offering right?

Reply:
I can extend this validation to also include it.

Reply:
Added the new offering intensity validation.